services/aster-nix/src/net/iface/util.rs

```diff
@@ -46,30 +48,45 @@ impl BindPortConfig {
 }
 
 pub fn spawn_background_poll_thread(iface: Arc<dyn Iface>) {
-    // FIXME: use timer or wait_timeout when timer is enable.
     let task_fn = move || {
-        debug!("spawn background poll thread");
+        trace!("spawn background poll thread for {}", iface.name());
+        let wait_queue = iface.polling_wait_queue();
         loop {
-            let next_poll_time = if let Some(next_poll_time) = iface.next_poll_at_ms() {
-                next_poll_time
+            let next_poll_at_ms = if let Some(next_poll_at_ms) = iface.next_poll_at_ms() {
+                next_poll_at_ms
             } else {
-                Thread::yield_now();
-                continue;
+                wait_queue.wait_until(|| iface.next_poll_at_ms())
             };
-            let now = read_monotonic_milli_seconds();
-            if now > next_poll_time {
-                // FIXME: now is later than next poll time. This may cause problem.
+
+            let now_as_ms = read_monotonic_milli_seconds();
+
+            // FIXME: Ideally, we should perform the `poll` just before `next_poll_at_ms`. However, this
+            // approach may result in a spinning busy loop if the `poll` operation yields no results. To
+            // mitigate this issue, we have opted to assign a high priority to the polling thread, ensuring
+            // that the `poll` runs as soon as possible.
+            // For a more in-depth discussion, please refer to the following link:
+            // https://github.com/asterinas/asterinas/pull/630#discussion_r1496817030.
+            if now_as_ms >= next_poll_at_ms {
                 iface.poll();
                 continue;
             }
-            let duration = next_poll_time - now;
-            // FIXME: choose a suitable time interval
-            if duration < 10 {
-                iface.poll();
-            } else {
-                Thread::yield_now();
-            }
+
+            let duration = Duration::from_millis(next_poll_at_ms - now_as_ms);
+            wait_queue.wait_until_or_timeout(
+                || {
+                    let next_poll_at_ms = iface.next_poll_at_ms()?;
+                    let now_as_ms = read_monotonic_milli_seconds();
+                    if next_poll_at_ms <= now_as_ms {
+                        Some(())
+                    } else {
+                        None
+                    }
+                },
```

Comment:
Well, this still does not work properly. For example, suppose we are at 10:00 and `next_poll_at_ms` is at 10:30. Then at 10:15, `next_poll_at_ms` gets advanced from 10:30 to 10:20. In this case we need to break from `wait_queue.wait_until_or_timeout` and set a new duration to wait, otherwise the thread will be woken at 10:30 (old `next_poll_at_ms`) instead of 10:20 (new `next_poll_at_ms`).

I think we need to follow my original suggestion in [this comment](https://github.com/asterinas/asterinas/pull/630#discussion_r1496808232), unless there is a better *correct* solution:
```suggestion
                // If changes in `next_poll_at_ms` invalidate the `duration` to wait,
                // we will end the wait and retry.
                || (iface.next_poll_at_ms() != Some(next_poll_at_ms)).then_some(()),
```

Reply:
Thanks for your comment!

I change the condition as 
```rust
(iface.next_poll_at_ms()? < next_poll_at_ms).then_some(())
```
The wait will be ended if `iface.next_poll_at_ms()` changes to an earlier time.

Reply:
Yes, it makes sense.